icp-deploy.tf

```diff
@@ -64,11 +68,13 @@ module "icpprovision" {
     # We will let terraform generate a new ssh keypair
     # for boot master to communicate with worker and proxy nodes
     # during ICP deployment
-    generate_key = true
+    generate_key    = false
+    icp_pub_key     = "${local.icp_pub_key}"
+    icp_priv_key    = "${local.icp_priv_key}"
 
     # SSH user and key for terraform to connect to newly created VMs
     # ssh_key is the private key corresponding to the public assumed to be included in the template
-    ssh_user        = "${var.ssh_user}"
-    ssh_key_base64  = "${base64encode(file(var.ssh_keyfile))}"
+    ssh_user        = "${local.ssh_user}"
```

Comment:
I don't really understand why we're using local.ssh_user here, since line 7 in the same file declares local.ssh_user = var.ssh.  Locals are normally used to make code easier to read, by moving complex interpolations and logic to a common place so as to avoid terribly long lines and unreadable code. Not sure the benefit in this case?

Reply:
@hassenius no benefit really. I think I was simply trying to stay consistent to what was in the file already.